sentinelsat/sentinel.py

```diff
@@ -1097,6 +1097,44 @@ def check_bounds(geometry):
     wkt = re.sub(r'(?<!\d) ', '', wkt)
     return wkt
 
+def placename_to_wkt(placename):
+    """Maps the placename to rectangular bounding extents using Nominatim API and converts to Well-Known-Text.
+   
+    Parameters
+    ----------
+    placename : str
+        the query to geocode
+    
+    Returns
+    -------
+    polygon coordinates
+        string of comma separated coordinate tuples (lon, lat) to be used by SentinelAPI
+    """
+
+    rqst = requests.post('https://nominatim.openstreetmap.org/search', \
+                         params = { 'q':placename,'format':'geojson'})
+    #Check that the response from Openstreetmapserver has status code 2xx 
+    #and that the response is valid JSON.
+    _check_scihub_response(rqst)
+    jsonlist = rqst.json()
+    try:
+        #Get the First result's bounding box.
+        bbox = jsonlist['features'][0]['bbox']
+    except IndexError:
+        #bbox is empty
+        raise Exception("No match found for entered Placename query!")
+    except:       
+        #unknown error
+        raise Exception("Unexpected error")
+    else:
+        #the bbox follows the pattern:
+        #[longitude lowerleft corner,latitude lowerleft corner,longitude upperright corner,latitude upperright corner]
+        footprint = "POLYGON(({} {},{} {},{} {},{} {},{} {}))"
```

Comment:
SciHub's Solr backend supports a much simpler `ENVELOPE(minX, maxX, maxY, minY)` WKT geometry, I'd suggest using it instead of `POLYGON((...))`. See [this test](https://github.com/sentinelsat/sentinelsat/blob/d9333fd92fc1fdf182bf0f55da34412586427ff3/tests/test_opensearch.py#L325) for an example.
Also, as a side note, `footprint:[minY,minX TO maxY,maxX]` works as an equivalent to `footprint:"Intersects(ENVELOPE(minX, maxX, maxY, minY))"`, curiously.

Reply:
I would also unpack the `bbox` variable as `minx, miny, maxx, maxy = bbox` and use these to construct the WKT string for clarity.

Reply:
i will make the changes accordingly